server/services/ewelink/lib/device/connect.js

```diff
@@ -0,0 +1,61 @@
+const Promise = require('bluebird');
+const { EVENTS, WEBSOCKET_MESSAGE_TYPES } = require('../../../../utils/constants');
+const { ServiceNotConfiguredError } = require('../../../../utils/coreErrors');
+const { EWELINK_EMAIL_KEY, EWELINK_PASSWORD_KEY, EWELINK_REGION_KEY, EWELINK_REGIONS } = require('../utils/constants');
+
+/**
+ * @description Connect to eWeLink cloud account and get access token and api key.
+ * @example
+ * connect();
+ */
+async function connect() {
+  this.configured = false;
+  this.connected = false;
+
+  /* eslint-disable prefer-const */
+  let [email, password, region] = await Promise.all([
+    this.gladys.variable.getValue(EWELINK_EMAIL_KEY, this.serviceId),
+    this.gladys.variable.getValue(EWELINK_PASSWORD_KEY, this.serviceId),
+    this.gladys.variable.getValue(EWELINK_REGION_KEY, this.serviceId),
+  ]);
+  /* eslint-enable prefer-const */
+
+  if (!email || !password) {
+    this.gladys.event.emit(EVENTS.WEBSOCKET.SEND_ALL, {
+      type: WEBSOCKET_MESSAGE_TYPES.EWELINK.ERROR,
+      payload: 'Service is not configured',
+    });
+    throw new ServiceNotConfiguredError('EWeLink error: Service is not configured');
+  }
+
+  if (!Object.values(EWELINK_REGIONS).includes(region)) {
+    const connection = new this.EweLinkApi({ email, password });
+    const response = await connection.getRegion();
+    // belt, suspenders ;)
+    if (response.error && [401, 406].indexOf(response.error) !== -1) {
+      response.msg = 'Service is not configured';
+    }
+    await this.throwErrorIfNeeded(response, true, true);
+
+    ({ region } = response);
```

Comment:
So here could be `region = response.region`.

Reply:
No, because of the eslint rule : [prefer-destructuring](https://eslint.org/docs/rules/prefer-destructuring)